esmvalcore/preprocessor/_io.py

```diff
@@ -134,6 +157,50 @@ def load(
     return cubes
 
 
+def _load_zarr(
+    file: str | Path | Cube | CubeList | xr.Dataset | ncdata.NcData,
+    ignore_warnings: list[dict[str, Any]] | None = None,
+    backend_kwargs: dict[str, Any] | None = None,
+) -> CubeList:
+    if isinstance(file, Path):
+        zarr_xr = xr.open_dataset(
+            file,
+            consolidated=False,
+            engine="zarr",
+        )
+    elif urlparse(file).scheme in ["http", "https"]:
+        # basic test that opens the Zarr/.zmetadata file for Zarr2
+        # or Zarr/zarr.json for Zarr3
+        fs = fsspec.filesystem("http")
+        zarr2 = zarr3 = True
+        try:
+            fs.open(file + "/.zmetadata", "rb")  # Zarr2
+        except Exception:  # noqa: BLE001
+            zarr2 = False
+        try:
+            fs.open(file + "/zarr.json", "rb")  # Zarr3
+        except Exception:  # noqa: BLE001
+            zarr3 = False
+        # we don't want to catch any specific aiohttp/fsspec exception
+        # bottom line is that that file has issues, so raise
+        if not zarr2 and not zarr3:
+            msg = f"File '{file}' can not be open as Zarr file at the moment."
+            raise ValueError(msg) from None
```

Comment:
What happens if you try to read files that are not Zarr2 or Zarr3? Will this raise an error? If yes, I would simply let the underlying code raise that error and not build in this logic here.

Reply:
there are no other Zarr files than Zarr2 and Zarr3: Zarr used to be just Zarr, before the major super release of 3.0, that contained a lot of structural changes, also a lot of breaking changes, after that it's sort of lore that people call Zarr2 and Zarr3 to differentiate between the two formats, it's not an official format per se, it's still "Zarr". Whatever it's not Zarr is caught by this test, also this catches issues with the network or temporary unavailability of the object store (that's why I phrased the exception "at the moment" :wink: ). Here is the Zarr3 mega release (note that they had some serious issues with 3.1, so they had to yank it) https://zarr.dev/blog/zarr-python-3-release/

Reply:
All right, but what happens if a non-Zarr2/non-Zarr3 is passed to `xr.open_dataset`. I would hope that this raises an error. In that case, I think we can remove all the code here that is just there to raise error?

Reply:
bingo! That's why the test here: Xarray uses Zarr that uses fsspec and aiohttp to access files over the network, and indeed you will get an aiohttp error if one tries to access a Zarr file that is not Zarr, but that error usually comes after a good few minutes of waiting, and with a huge stracktrace - we want to avoid exactly that and literally poke the file before trying to open it. Telling you, man, it's only a few lines of code in this PR, but that's been a quite a bit of seat and guts before these :grin:

Reply:
it is, even cf-python stole this test :grin: